packages/contract/src/builder.test-d.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  defineConsumer,
+  defineContract,
+  defineExchange,
+  defineMessage,
+  definePublisher,
+  defineQueue,
+  defineQueueBinding,
+  mergeContracts,
+} from "./builder.js";
+import { describe, expectTypeOf, it } from "vitest";
+import type { ContractDefinition } from "./types.js";
+import { z } from "zod";
+
+describe("mergeContracts - Type Tests", () => {
+  describe("Type safety and inference", () => {
+    it("should return a valid ContractDefinition when merging two contracts with publishers", () => {
+      // GIVEN
+      const orderExchange = defineExchange("orders", "topic", { durable: true });
+      const paymentExchange = defineExchange("payments", "topic", { durable: true });
+
+      const orderMessage = defineMessage(z.object({ orderId: z.string() }));
+      const paymentMessage = defineMessage(z.object({ paymentId: z.string() }));
+
+      const orderContract = defineContract({
+        exchanges: { orders: orderExchange },
+        publishers: {
+          orderCreated: definePublisher(orderExchange, orderMessage, {
+            routingKey: "order.created",
+          }),
+        },
+      });
+
+      const paymentContract = defineContract({
+        exchanges: { payments: paymentExchange },
+        publishers: {
+          paymentReceived: definePublisher(paymentExchange, paymentMessage, {
+            routingKey: "payment.received",
+          }),
+        },
+      });
+
+      // WHEN
+      const merged = mergeContracts(orderContract, paymentContract);
+
+      // THEN - Verify result is a valid ContractDefinition
+      expectTypeOf(merged).toMatchTypeOf<ContractDefinition>();
+      expectTypeOf(merged).toHaveProperty("exchanges");
+      expectTypeOf(merged).toHaveProperty("publishers");
```

Comment:
not enought, we should assert what is in exchanges in oder to make sure merge has worked correctly. same for publishers